### PR TITLE
Add VLAN tagged packet size test

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Add  `test_vlan_tagged_packet_size` test
Signed-off-by: Stepan Vovk <stepan.vovk@plvision.eu>

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    streams = {f"Untagged traffic": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "src_mac": "02:00:00:00:00:01",
+        "dst_mac": "ff:ff:ff:ff:ff:ff",
+        "frameSize": 512
+        }}
+
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py]:
```suggestion
        "frameSize": 512,
    }}

```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    # 6. Verify that the tagged packet size is bigger in 4 bytes than untagged packet
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    frame_size = streams["Untagged traffic"].get("frameSize")
+    for row in stats.Rows:
+        if row["Rx Port"] == tg_ports[tagged_traffic_port]:
+            tagged_traffic_size = int(row["Rx Bytes"]) / int(row["Rx Frames"]) - frame_size
+        else:
+            untagged_traffic_size = int(row["Rx Bytes"]) / int(row["Rx Frames"]) - frame_size
+    assert tagged_traffic_size - untagged_traffic_size == 4, f"Expected difference in 4 bytes"
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py]:
```suggestion
    expected_packet_size = streams["Untagged traffic"].get("frameSize")
    for row in stats.Rows:
        actual_packet_size = int(row["Rx Bytes"]) / int(row["Rx Frames"])
        if row["Rx Port"] == tg_ports[tagged_traffic_port]:
            assert actual_packet_size == expected_packet_size + 4 \
                f"Expected packet size to be {expected_packet_size + 4}, but actual size is {actual_packet_size}"
        else:
            assert actual_packet_size == expected_packet_size \
                f"Expected packet size to be {expected_packet_size}, but actual size is {actual_packet_size}"
```

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

**Review by SerhiyBoikoPLV — APPROVED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            {"port": 1, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 2, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 3, "settings": [{"vlan": 1, "untagged": False, "pvid": True}]},)
+
+
+async def test_vlan_tagged_packet_size(testbed):
+    """
+    Test Name:VLAN tagged packet size
+    Test Suite: suite_functional_vlan
+    Test Overview: Test packet with VLAN tag on untagged port is bigger by 4 bytes
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py]: How can we observe tagged packets on an untagged port? Please correct this statement.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(5)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # 6. Verify that the tagged packet size is bigger in 4 bytes than untagged packet
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    expected_packet_size = streams["Untagged traffic"].get("frameSize")
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py]: "actual" and "expected" are a little bit confusing. Especially when you compare with expected + 4. I suggest to use tx_frame_size and rx_frame_size.

**Review by anton7811 — CHANGES_REQUESTED** [review]

**Review by anton7811 — APPROVED** [review]

**Pull request closed without merging**